### ISSUE TITLE
Add remaining regions to Tanzania

### ISSUE THREAD

**Issue by sharpphyl** [ArctosDB/arctos#5395]
Arctos geography shows 25 entries for Tanzania.
<img width="618" alt="Screenshot 2022-12-20 at 9 04 30 AM" src="https://user-images.githubusercontent.com/15368365/208711876-b9a6727c-f8f7-4549-9be5-bfad71d9516d.png">

GADM shows 30 regions for Tanzania

<img width="735" alt="Screenshot 2022-12-20 at 9 05 38 AM" src="https://user-images.githubusercontent.com/15368365/208711966-a394d9fb-b2dc-4537-816d-5f6e404a9cb2.png">

We're missing at least three Zanzibar regions which I need for databasing.

**Comment by sharpphyl** [comment]
I'm finding more.  The Bahamas in Arctos shows 14 entries (including Bahamas)

<img width="847" alt="Screenshot 2022-12-20 at 9 14 54 AM" src="https://user-images.githubusercontent.com/15368365/208714001-67787cc0-dfac-4b32-badd-915787583d11.png">

GADM shows 32 first level subdivisions.

<img width="817" alt="Screenshot 2022-12-20 at 9 15 05 AM" src="https://user-images.githubusercontent.com/15368365/208714089-418a5b39-37de-4a3b-8f05-2be655cb82ca.png">

Shouldn't we see all the first level subdivisions in all these countries?

**Comment by dustymc** [comment]
>Shouldn't we see all the first level subdivisions in all these countries?

If that's the request it needs to be made clear, and probably accompanied by physical resources - I'm already struggling to do anything with the mass of data we have at present.

If you need something specific then I need to know what it is.

**Comment by sharpphyl** [comment]
> If  you need something specific then I need to know what it is.

Do you need a csv from me for these?

To clear up the locality notifications, we need Zanzibar North, Zanzibar West, Zanzibar South and Central in Tanzania.

Bahamas:  North Eleuthera, Central Eleuthera, South Eleuthera, North Abaco, Central Abaco, South Abaco, East Grand Bahama, West Grand Bahama, Long Island, Hope Townm and Grand Cay.  GADM shows the province as "Biminis" rather than Bimini.

We need the missing subdivisions in Japan: Akita, Ishikawa, Toyama, Kagawa, and Saga. 

And Madagascar seems to be the only country still associated with a continent and with an Island.  The subdivisions are fine.

<img width="1055" alt="Screenshot 2022-12-20 at 9 18 20 AM" src="https://user-images.githubusercontent.com/15368365/208714596-15b7f039-256c-4cc6-8889-e6a2cbb85caf.png">

I'll file separate issues if we find others we need.

**Comment by dustymc** [comment]
@sharpphyl I'm still trying to figure out how to do this from the current state of things, but i think https://github.com/ArctosDB/arctos/issues/5376 has been illuminating so here's some comments and CSV.

>To clear up the locality notifications,

Can you elaborate on that? I'm not understanding how it's related, and it might change the priority on dealing with all appropriate data.

I think I want to try a "name the country and I'll provide CSV" approach to this - that might be a little spotty but everything else is leading to some outright black hole from which no information can escape... That's also going to require some cleanup, but that needs done anyway. So - let's try.

>I'll file separate issues if we find others we need.

Yes, I'll add that to the Issue template - I think I have to do this one entity at a time or I'm going to get lost, as I hopefully won't demonstrate here!


> province as "Biminis" rather than Bimini.

You should be able to do that. The names/terms matching is super important to the approach I'm going to try, so please either clean up or let me know if you see some mismatch. I fixed a few more in the Bahamas, but could have still missed something.

>Madagascar


Madagascar is https://github.com/ArctosDB/arctos/issues/5185 - I'm not sure what to do there, other than hoping @mkoo  has some line to GADM and we can get them to prioritize.



>Tanzania

I cleaned up the mess of current data, pulled what's in GADM, and deleted what we allegedly already have from the cleaned Arctos data. Please

1. For anything you want created, supply an appropriate wikipedia link. 
2. For anything you DO NOT want created, delete the row.
3. Let me know about any "would be duplicates" or other messes, somehow it seems we'll never get beyond the possibility but do what you can and I'll review before loading and hopefully we'll catch everything between us.

(And it would save some future struggle if we just created the whole country, wouldn't it? We need to somehow prioritize doing whatever it takes to be able to perform fast spatial operations on bigger-than-now datasets.)

[temp_geog_create(2).csv](https://github.com/ArctosDB/arctos/files/10271455/temp_geog_create.2.csv)

>Bahamas

Here you go, same drill as above.

[temp_geog_create(4).csv.zip](https://github.com/ArctosDB/arctos/files/10271797/temp_geog_create.4.csv.zip)

**Comment by sharpphyl** [comment]
> To  clear up the locality notifications,

> Can you elaborate on that? I'm not understanding how it's related, and it might change the priority on dealing with all appropriate data.

Here's a typical example: locality 10964382

<img width="1800" alt="Screenshot 2022-12-20 at 2 28 48 PM" src="https://user-images.githubusercontent.com/15368365/208770472-092f3dbb-4f65-406a-ba32-01db8b28d3ae.png">

The previous geography was Africa, Tanzania, Zanzibar, Zanzibar Archipelago, Unguja with a specific locality of Nunjwi.  The new geography is Tanzania.  A search shows that Nunjwi is at the far north of the Island Unguja so it would be in Zanzibar North subdivision, which wasn't yet in the Arctos list.  There are several others in that group of notifications from Zanzibar that would be more complete if we add the Zanzibar province.

Similarly with the Bahamas, Locality 10890400 had previous geography of West Indies, Bahamas, Bahama Islands, Eleuthera and a specific locality of Tarpum Bay.  There are three provinces on Eleuthera and the previous geography did not identify which one and the island field has been eliminated, so the new geography is just Bahamas.  Tarpum Bay is in South Eleuthera District which was missing from Arctos.  

> And it would save some future struggle if we just created the whole country, wouldn't it?

I wish I could suggest how to prioritize countries but I find missing subdivisions or mismatches by chance and I don't know what countries you've recently been updating until I open up the notifications and I have many, many more to go.  Plus we have no idea what's in our backlog.  It may be best for me to just use GitHub when I find needed geography for now unless other collections have similar needs.

I modified the Tanzania csv and will work on the Bahamas tomorrow.  A bit of a mess in several ways.  I'm assuming you want to use the name of the subdivision that is in GADM if it conflicts with Wikipedia.  

The first five entries appear to be alternate search terms. They are all Swahili for the region names.  I would prefer to have them in English as that is the way we usually search for them and have the Swahili as an alternate search term.  I have added the English terms in the last five lines. I would delete the first five but wanted to check first to make sure that they aren't something we need.  

The next two are fine.  I've added five more regions.  You'll see that GADM and Wikipedia don't agree on the primary term to use for the region names but they are the same.  I don't know where your TZ come from under SEARCH_TERMS so you'll see a lot of ???? there.

Let me know if I should make any changes in how I review the Bahamas csv next.  Thanks!

[temp_geog_create.2.csv](https://github.com/ArctosDB/arctos/files/10273191/temp_geog_create.2.csv)

**Comment by dustymc** [comment]
>I'm assuming you want to use the name of the subdivision that is in GADM if it conflicts with Wikipedia.

Yes, GADM (the shape, not necessarily the terms) should be seen as authoritative (because it is), wiki just helps people understand what might be going on  there. (I'd sort of intended to get rid of the wiki requirement altogether, but it's been REALLY useful a bunch of time so I think I still like it, but always up for clever ideas.)

>sure that they aren't something we need.

They are; our model is "do what GADM does" and everything I'm building to some extent arises from the idea that we'll actually do that and not wander off on our own tangents. If they want to call something A and absolutely everybody else wants to call it Z then A still wins, at least until we find another model (and that might drive us to start looking...).

>. I don't know where your TZ come from under SEARCH_TERMS 

Various stuff, but that's just search terms - feel free to remove them or add lots more (it's a pipe-delimited list).

**Comment by sharpphyl** [comment]
> They are; our model is "do what GADM does"

It appears you're working from a different list than what is online at https://gadm.org/maps/TZA.html

I can't find the first five names (lines 2-6) in GADM as first level subdivisions of Tanzania under Maps.  I had to google them to figure out what they were.  I don't recommend we add them except as additional search terms in Swahili. 

I would add the six subdivisions on lines 8-13 which are listed at https://gadm.org/maps/TZA_1.html.  We should also add the Songwe Region on line 7 although it is missing in GADM - but you had it in the original csv so, again, you must have a different list.  (It was created in 2016 out of Mybeya.) 

That would give us 31 subdivisions which is accurate per Wikipedia.  

> province as "Biminis" rather than Bimini.
> You should be able to do that.

I don't think I have authority to make any changes to geography.

**Comment by sharpphyl** [comment]
And here are the Bahamas.  Much cleaner.
[temp_geog_create(4) pms edits.csv](https://github.com/ArctosDB/arctos/files/10278306/temp_geog_create.4.pms.edits.csv)

**Comment by dustymc** [comment]
>working from a different list than what is online 

Yes, they don't seem to build their webpage from their data and sometimes they're different. The dates/version aren't clear, but the data are 4.1 which seems to be summer 2022 and the map claims to be © 2018 GADM so I'd guess i've got the most-current, but I'm not at all sure of that. I can renew and see what happens.....

```
 gadm1::Tanzania|Arusha
 gadm1::Tanzania|Dar es Salaam
 gadm1::Tanzania|Dodoma
 gadm1::Tanzania|Geita
 gadm1::Tanzania|Iringa
 gadm1::Tanzania|Kagera
 gadm1::Tanzania|Kaskazini Pemba
 gadm1::Tanzania|Kaskazini Unguja
 gadm1::Tanzania|Katavi
 gadm1::Tanzania|Kigoma
 gadm1::Tanzania|Kilimanjaro
 gadm1::Tanzania|Kusini Pemba
 gadm1::Tanzania|Kusini Unguja
 gadm1::Tanzania|Lindi
 gadm1::Tanzania|Manyara
 gadm1::Tanzania|Mara
 gadm1::Tanzania|Mbeya
 gadm1::Tanzania|Mjini Magharibi
 gadm1::Tanzania|Morogoro
 gadm1::Tanzania|Mtwara
 gadm1::Tanzania|Mwanza
 gadm1::Tanzania|Njombe
 gadm1::Tanzania|Pwani
 gadm1::Tanzania|Rukwa
 gadm1::Tanzania|Ruvuma
 gadm1::Tanzania|Shinyanga
 gadm1::Tanzania|Simiyu
 gadm1::Tanzania|Singida
 gadm1::Tanzania|Songwe
 gadm1::Tanzania|Tabora
 gadm1::Tanzania|Tanga
```

I think that's the same as what I had, but let me know if you want me to csv-create tablify it.

>accurate per Wikipedia

https://en.wikipedia.org/wiki/Regions_of_Tanzania seems to line up with what I have? if there's a difference, I think it usually comes down to GADM being out of date (in which you might wish to file a "geography cleanup" issue) or GADM trying to represent the place as the locals see it, while wikipedia usually reflects the views of the US+EU.

**Comment by dustymc** [comment]
>Bahamas

Done, thanks.

**Comment by sharpphyl** [comment]
> GADM trying to represent the place as the locals see it, while wikipedia usually reflects the views of the US+EU.

I think that's the reason for the differences between GADM 2022 and Wikipedia.  The difference between your GADM download and their online data is probably just their delay in synchronizing both resources.  I can file an issue through their "contact us" form.  

Let's go ahead and upload the gadm1 [above](https://github.com/ArctosDB/arctos/issues/5395#issuecomment-1361571716).  The additional entries would be the lines 2-8 in the last csv I sent.  I'll check after the upload to see if we need to add any additional search terms for the US+EU terms. Let me know if you need anything else from me.  Thanks.

**Comment by dustymc** [comment]
I'm sorry if I'm being especially dense today, but I'm not absolutely sure what I'm supposed to do. Here's the current Tanzania from the most recent GADM data which aren't in Arctos:

[temp_geog_create(6).csv](https://github.com/ArctosDB/arctos/files/10279709/temp_geog_create.6.csv)

Could you please fill (with wikipedia articles) source_authority, or if you've already done that in another file reattach it here?

**Comment by sharpphyl** [comment]
No problem.  My comment was convoluted.  Here's the csv with source_authority.  

[temp_geog_create.6.csv](https://github.com/ArctosDB/arctos/files/10280703/temp_geog_create.6.csv)
